docs/reference/search/suggesters/context-suggest.asciidoc

```diff
@@ -1,237 +1,269 @@
 [[suggester-context]]
 === Context Suggester
 
-The context suggester is an extension to the suggest API of Elasticsearch. Namely the
-suggester system provides a very fast way of searching documents by handling these
-entirely in memory. But this special treatment does not allow the handling of
-traditional queries and filters, because those would have notable impact on the
-performance. So the context extension is designed to take so-called context information
-into account to specify a more accurate way of searching within the suggester system.
-Instead of using the traditional query and filter system a predefined ``context`` is
-configured to limit suggestions to a particular subset of suggestions.
-Such a context is defined by a set of context mappings which can either be a simple
-*category* or a *geo location*. The information used by the context suggester is
-configured in the type mapping with the `context` parameter, which lists all of the
-contexts that need to be specified in each document and in each suggestion request.
-For instance:
+The completion suggester considers all documents in the index, but it is often
+desirable to serve suggestions filtered and/or boosted by some criteria.
+For example, you want to suggest song titles filtered by certain artists or
+you want to boost song titles based on their genre.
+
+To achieve suggestion filtering and/or boosting, you can add context mappings while
+configuring a completion field. You can define multiple context mappings for a
+completion field.
+Every context mapping has a unique name and a type. Currently, there are two types
+*category* and *geo*. The context mappings is configured under the `contexts` parameter
```

Comment:
The context mappings ->

    Context mappings are configured with the `contexts` parameter in the field mapping.